EDU-10762 - Updating Building a Shelf doc (#1181)
vtexdocs/dev-portal-content: docs/guides/vtex-io/Storefront-Guides/concepts-1/vtex-io-documentation-building-a-shelf.md

slug: "vtex-io-documentation-building-a-shelf"
hidden: false
createdAt: "2021-10-28T00:08:33.278Z"
updatedAt: "2024-05-28T13:00:00.565Z"
---

The Shelf component is a product list that helps you to build your own shop window and work on your store's visual merchandising.

![shelf](https://cdn.jsdelivr.net/gh/vtexdocs/dev-portal-content@main/images/vtex-io-documentation-building-a-shelf-0.png)

> Aiming to display a flexible product list in your home page, the Shelf component is configured using the [Product Summary List](https://developers.vtex.com/docs/guides/vtex-product-summary-productsummarylist), the [Product Summary Shelf](https://developers.vtex.com/docs/guides/vtex-product-summary/), and the [Slider Layout](https://developers.vtex.com/docs/guides/vtex-slider-layout/) blocks, which can be combined to create flexible and dynamic product lists.

Reviewer: ```suggestion
To create a product list, you can configure the Shelf component to use the [Product Summary List](https://developers.vtex.com/docs/apps/vtex.product-summary/productsummarylist), the [Product Summary Shelf](https://developers.vtex.com/docs/apps/vtex.product-summary), and the [Slider Layout](https://developers.vtex.com/docs/apps/vtex.slider-layout) blocks.
```

Reviewer: Esse parágrafo está bem wordy. Fiz uma sugestão para facilitar a leitura e atualizei os links

> >ℹ️ Since these three blocks come from the [**Product Summary**](https://github.com/vtex-apps/product-summary) and the [**Slider-Layout**](https://github.com/vtex-apps/slider-layout) app, it is strongly recommended to check out each documentation in order to understand how the blocks work as well as the props available to set their behavior and consequently the Shelf behavior.

Reviewer: ```suggestion
>ℹ️ Refer to the [**Product Summary**](https://developers.vtex.com/docs/apps/vtex.product-summary) and [**Slider-Layout**](https://developers.vtex.com/docs/apps/vtex.slider-layout) documentation to explore the available props for configuring their behavior. Use this information to adjust the Shelf behavior accordingly.
```

Reviewer: Esse parágrafo também está bem wordy. Sugeri quebrar em sentenças mais curtas e atualizei os links

  },

  "product-summary.shelf#demo1": {
    "children": [
      "product-summary-name",
      "product-summary-description",
      "product-summary-image",
      "product-summary-price",
      "product-summary-sku-selector",
      "product-summary-buy-button"
    ]

>   }

Reviewer: Repara que o esse código está configurado com `diff`. Esses `+`/`-` servem para destacar o que muda/o que sai qnd é um `diff`. Se você optar por apresentar direto como json, não deixe de atualizar o bloco de código para `json` para que ele seja destacado corretamente. Exemplo:

- DIFF
```diff
- { 
+  "product-summary.shelf#demo1": 
+   "children": [	    
+    "product-summary-name",	      
+    "product-summary-description",	     
+    "product-summary-image",	    
+    "product-summary-price",	
+    "product-summary-sku-selector",	    
+    "product-summary-buy-button"	    
+   ]	
+ }
```

- JSON
```json
{
  "product-summary.shelf#demo1": {
    "children": [
      "product-summary-name",
      "product-summary-description",
      "product-summary-image",
      "product-summary-price",
      "product-summary-sku-selector",
      "product-summary-buy-button"
    ]
  }
```
